### ISSUE TITLE
MD034 False positive for certain link markup

### ISSUE THREAD

**Issue by brlin-tw** [markdownlint/markdownlint#328]
## Description
The following markup:

```markdown
# title

* [從零開始的 Telegram Bot | Sean's Note](https://blog.sean.taipei/2017/05/telegram-bot)

```

triggers the following false positive:

```
$ mdl testcase.md
testcase.md:1: MD034 Bare URL used

```

(notice the wrong line number indicator)

I managed to further reduce the reproducing case to:

```markdown
# title

* [|'](https://example.com)

```

### Environment
Also reproduced in an Alpine container from scratch

**MDL Version**
0.9.0 via RVM gem

**Ruby Version**
2.7.0p0 via RVM

**OS Version**
Ubuntu 20.04 AMD64

**Locale**
zh_TW.UTF-8

**Custom Configuration**
None

### Expected Behavior
No errors

**Comment by nightlark** [comment]
This bug also gets triggered by markdown wrapped with in-line html such as:
```
<details>
#### [zmq](http://www.zeromq.org)

some text [LGPL](https://github.com/zeromq/libzmq/blob/master/COPYING.LESSER) more text [this github issue](https://github.com/zeromq/libzmq/issues/3787). blah
</details>
```

Removing the html details tags surrounding the markdown makes the error go away. The [nodejs markdownlint](https://github.com/DavidAnson/markdownlint) does not have this bug.

**Comment by janosh** [comment]
Also encountering the @nightlark's type of false positives, i.e. non-bare links inside HTML tags being flagged as bare.

**Comment by nbehrnd** [comment]
@brlin-tw I'm guessing -- so you might want to add some ropes here -- the rendering of the markdown as a .html would be similar to the following:

![example](https://github.com/markdownlint/markdownlint/assets/19709084/81d7b37e-8706-4078-aa2a-603dc6227a17)

Inspired by some work with a regular expression, I assumed the pipe character would be a special one (for its function to define a table's column) and hence would require an escape to be processed as plain character only.  This is the reason why my «source code» for the above is

```markdown
# title

* [從零開始的 Telegram Bot \| Sean's Note](https://blog.sean.taipei/2017/05/telegram-bot)

## title 2, lightly edited

* [\|'](https://example.com)
```

If submitted to mdl (version 0.13.0, packaged as 0.13.0-3 for Debian 13/trixie, branch testing, [tracker](https://tracker.debian.org/pkg/ruby-mdl)), there is no error reported.  In addition to this, pandoc (version 3.1.6 by the project's [release page](https://github.com/jgm/pandoc/releases)) processes the .md to yield the above by

```shell
pandoc -s -i modified.md -o modified.html
```

[data.zip](https://github.com/markdownlint/markdownlint/files/13384522/data.zip)

**Comment by nbehrnd** [comment]
@nightlark As soon as there is either `<ftp:/details>`, or `<http:/details>` instead of `<details>`, the suggestions by mdl are gone.  There seems no need to use a `//` (as in `https://example.com`) to obtain an improvement *for checks by mdl*. 

One test used the following input:

```markdown
# principal header

<ftp:/details>

#### [zmq](http://www.zeromq.org)

some text [LGPL](https://github.com/zeromq/libzmq/blob/master/COPYING.LESSER)
more text [this github issue](https://github.com/zeromq/libzmq/issues/3787). blah
</details>
```

[data_nightlark.zip](https://github.com/markdownlint/markdownlint/files/13384628/data_nightlark.zip)

**Comment by nightlark** [comment]
@nbehrnd I’m not sure I follow… are you just providing more information about the bug, or are you suggesting that I use `<ftp:/details>` instead of `<details>`? `<details>` is an HTML tag with a meaning that affects how the document is displayed, not a bare url.

If its the latter, then your suggestion doesn’t work because `ftp:/details` isn’t recognized as a substitute for the `details` html tag, and no collapsible section gets created. The use of ftp:/ or http:/ isn’t improving anything, it’s breaking the rendering of the document and altering the intent of that being an html tag, not a link to some “details” website.

with `<details>` (what it should look like):
<details>
some random info
</details>

with `<ftp:/details>` (not rendered correctly):
<ftp:/details>
some more random info
</details>

**Comment by nbehrnd** [comment]
@nightlark Oh; I see.

So far, the examples compiled by [learnxinyminutes](https://learnxinyminutes.com/docs/markdown/) provided enough tools to document with markdown.  They do not cover the concept of *collapsible sections* GitHub describes as an advanced feature ([ref](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections)) of the language.  Thus, I misunderstood your example as an application of a direct/ auto-link (`<foo@bar.com>` to yield <foo@bar.com>)

In this perspective, I agree my comment to you does not provide an improvement.  I'm going to retract it.
